api/fittings/lisk_params_validator.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+var _ = require('lodash');
+
+module.exports = function create (fittingDef, bagpipes) {
+
+	return function lisk_params_validator (context, cb) {
+
+		var error = null;
+
+		// todo: add support for validating accept header against produces declarations
+		// see: http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html
+		//var accept = req.headers['accept'];
+		//var produces = _.union(operation.api.definition.produces, operation.definition.produces);
+
+		if (context.request.swagger.operation) {
+			var validateResult = context.request.swagger.operation.validateRequest(context.request);
+			if (validateResult.errors.length) {
+				error = new Error('Validation errors');
+				error.statusCode = 400;
+
+				error.errors = _.map(validateResult.errors, function (e) {
+					var errs = _.pick(e, ['code', 'message', 'in', 'name', 'errors']);
+					errs.errors = _.map(e.errors, function (e){ return _.pick(e, ['code', 'message'] ); });
+					return errs;
+				});
+			}
+		} else {
+			console.log('not a swagger operation, will not validate response');
```

Comment:
How about using logger here?

Reply:
Two reasons left here. Can't find any way to pass our application scope logger down to individual pipe definition. And secondly this condition is never going to meet, as all our endpoints will have swagger based operation defined in yml. So I didn't invest more time to invent a way to pass our variables to pipe definition.

Reply:
We don't allow console logs in production, so this will never be seen anyways. ESLINT might get mad as well.

Reply:
Done.